petab/observables.py

```diff
@@ -79,3 +81,65 @@ def get_output_parameters(observable_df: pd.DataFrame,
                 output_parameters.add(sym)
 
     return output_parameters
+
+
+def get_formula_placeholders(formula_string: str, observable_id: str,
+                             override_type: str) -> List[str]:
+    """
+    Get placeholder variables in noise or observable definition for the
+    given observable ID.
+
+    Arguments:
+        formula_string: observable formula
+        observable_id: ID of current observable
+        override_type: 'observable' or 'noise', depending on whether `formula`
+            is for observable or for noise model
+
+    Returns:
+        List of placeholder parameter IDs in the order expected in the
+        observableParameter column of the measurement table.
+    """
+    if not formula_string:
+        return []
+
+    if not isinstance(formula_string, str):
```

Comment:
does this happen? could be dangerous if only "string-like" objects are passed. this is intended to test for numerics, right? maybe then test explicitly.

Reply:
Typehint asks for `str`. Not expecting anything funny then :).